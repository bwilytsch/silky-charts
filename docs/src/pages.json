[
  {
    "title": "Getting Started",
    "path": "/",
    "component": "Introduction"
  },
  {
    "title": "Bar",
    "path": "bar",
    "component": "BarChart"
  }
]
