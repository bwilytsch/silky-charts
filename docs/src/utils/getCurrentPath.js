export default (url, path) => {
  const baseUrl = url.replace('/', '');
  const urlRegExp = new RegExp(`^\/${baseUrl}\/?$`);
  const replaceRegExp = new RegExp(`\/${baseUrl}\/?`);

  return path.match(urlRegExp) ? '/' : path.replace(replaceRegExp, '');
};
