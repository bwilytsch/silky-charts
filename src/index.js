export { default as Bar } from './Bar';
export { default as BarLine } from './BarLine';
export { default as StackedArea } from './StackedArea';

export { default as toSeries } from './utils/toSeries';
