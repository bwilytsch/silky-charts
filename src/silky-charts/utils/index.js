export { default as allDate } from './allDate';
export { default as drawGrid } from './drawGrid';
export { default as extendXPath } from './extendXPath';
export { default as getDivergence } from './getDivergence';
export { default as getXScale } from './getXScale';
export { default as getYScale } from './getYScale';
export { default as rotateXLabels } from './rotateXLabels';
export { default as themes } from './themes';
