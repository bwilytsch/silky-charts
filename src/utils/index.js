export { default as allDate } from './allDate';
export { default as appendStackedValues } from './appendStackedValues';
export { default as buildStack } from './buildStack';
export { default as bySeries } from './bySeries';
export { default as classify } from './classify';
export { debounce } from './debounce';
export { debounceImmediate } from './debounce';
export { default as drawGrid } from './drawGrid';
export { default as extendXPath } from './extendXPath';
export { default as getBaseColor } from './getBaseColor';
export { default as getHoverColor } from './getHoverColor';
export { default as getId } from './getId';
export { default as getLength } from './getLength';
export { default as getLineDataForSeries } from './getLineDataForSeries';
export { default as getMax } from './getMax';
export { default as getStackedMax } from './getStackedMax';
export { default as getSeries } from './getSeries';
export { default as getSize } from './getSize';
export { default as getXScale } from './getXScale';
export { default as getYScale } from './getYScale';
export { default as isValidDate } from './isValidDate';
export { default as palette } from './palette';
export { default as rotateXLabels } from './rotateXLabels';
export { default as setLineType } from './setLineType';
export { default as setupData } from './setupData';
export { default as toSingleForm } from './toSingleForm';
export { default as toStackedForm } from './toStackedForm';
export { default as valueFor } from './valueFor';
