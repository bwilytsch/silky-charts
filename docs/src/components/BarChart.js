import React from 'react';
import styled from 'styled-components';
import { Bar } from 'silky-charts';
import Chart from '../containers/Chart';
import BarData from '../markdown/bar-data.md';
import BarProps from '../markdown/bar-props.md';
import { InlineCode, Pre } from '../ui/code';
import { Table, Td } from '../ui/table';

const data = [
  { name: 'E', value: 0.12702 },
  { name: 'T', value: 0.09056 },
  { name: 'A', value: 0.08167 },
  { name: 'O', value: 0.07507 },
  { name: 'I', value: 0.06966 },
  { name: 'N', value: 0.06749 },
  { name: 'S', value: 0.06327 },
  { name: 'H', value: 0.06094 },
  { name: 'R', value: 0.05987 },
  { name: 'D', value: 0.04253 },
  { name: 'L', value: 0.04025 },
  { name: 'C', value: 0.02782 },
  { name: 'U', value: 0.02758 },
  { name: 'M', value: 0.02406 },
  { name: 'W', value: 0.0236 },
  { name: 'F', value: 0.02288 },
  { name: 'G', value: 0.02015 },
  { name: 'Y', value: 0.01974 },
  { name: 'P', value: 0.01929 },
  { name: 'B', value: 0.01492 },
  { name: 'V', value: 0.00978 },
  { name: 'K', value: 0.00772 },
  { name: 'J', value: 0.00153 },
  { name: 'X', value: 0.0015 },
  { name: 'Q', value: 0.00095 },
  { name: 'Z', value: 0.00074 },
];

const H3 = styled.h3`
  margin-top: 40px;
`;

const BarChart = () => (
  <>
    <h2>Bar Chart</h2>
    <Chart>
      <Bar data={data} responsive tooltip />
    </Chart>
    <H3>Data Definition</H3>
    <BarData components={{ pre: Pre }} />
    <H3>Props</H3>
    <BarProps components={{ inlineCode: InlineCode, table: Table, td: Td }} />
  </>
);

export default BarChart;
