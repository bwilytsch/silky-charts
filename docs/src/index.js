import React from 'react';
import { render } from 'react-dom';
import { Router, Redirect } from '@reach/router';
import Layout from './containers/Layout';
import Header from './containers/Header';
import Navigation from './containers/Navigation';
import Content from './containers/Content';
import Introduction from './components/Introduction';
import BarChart from './components/BarChart';
import pages from './pages.json';
import 'normalize.css';
import './styles.css';

const App = () => {
  const baseURL = '/silky-charts';
  const pathname = window.location.pathname;
  const currentPath = pathname.includes(baseURL)
    ? pathname.replace(baseURL, '')
    : '/';
  const isValidPathname =
    currentPath && pages.map(({ path }) => path).includes(currentPath);

  console.log(currentPath);

  const contentComponents = {
    Introduction,
    BarChart,
  };

  return (
    <Layout>
      <Header />
      <Navigation
        items={pages}
        baseURL={baseURL}
        initial={isValidPathname ? currentPath : '/'}
      />
      <Router primary={false}>
        {currentPath === '/' && (
          <Redirect from={currentPath} to={baseURL} noThrow />
        )}
        <Content path={baseURL}>
          {pages.map(({ component, path }, idx) => {
            const Component = contentComponents[component];
            return <Component key={idx} path={path} default={path === '/'} />;
          })}
        </Content>
      </Router>
    </Layout>
  );
};

render(<App />, document.getElementById('app'));
