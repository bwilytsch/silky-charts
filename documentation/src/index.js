import React from 'react';
import { render } from 'react-dom';
import { Router, Redirect } from '@reach/router';
import Layout from './containers/Layout';
import Header from './containers/Header';
import Navigation from './containers/Navigation';
import Content from './containers/Content';
import Introduction from './components/Introduction';
import BarChart from './components/BarChart';
import pages from './pages.json';
import 'normalize.css';
import './styles.css';

const App = () => {
  const pathname = window.location.pathname;
  const defaultPath = '/';
  const currentPath =
    pathname !== defaultPath ? pathname.replace(defaultPath, '') : pathname;
  const isInvalidPath = !pages
    .map(({ pathname }) => pathname)
    .includes(currentPath);

  const contentComponents = {
    Introduction,
    BarChart,
  };

  return (
    <Layout>
      <Header />
      <Navigation
        items={pages}
        initialPath={isInvalidPath ? defaultPath : currentPath}
      />
      <Content>
        <Router>
          {isInvalidPath && (
            <Redirect from={currentPath} to={defaultPath} noThrow />
          )}
          {pages.map(({ component, pathname }, idx) => {
            const Component = contentComponents[component];
            return (
              <Component
                key={idx}
                path={pathname}
                default={pathname === defaultPath}
              />
            );
          })}
        </Router>
      </Content>
    </Layout>
  );
};

render(<App />, document.getElementById('app'));
