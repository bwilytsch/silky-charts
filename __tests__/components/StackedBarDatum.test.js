import React from 'react';
import { render, fireEvent, cleanup } from 'react-testing-library';
import always from 'ramda/src/always';
import { StackedBarDatum } from '../../src/components';
import { buildStack } from '../../src/utils';
import 'jest-styled-components';

const stack = buildStack(['bar', 'baz']);
const series = stack([
  { name: 'foo', foo: 0, bar: 1, baz: 2, qux: 3 },
  { name: 'baz', foo: 4, bar: 5, baz: 6, qux: 7 },
]);

const fn = always(0);
const xScale = fn;
xScale.bandwidth = fn;

const props = {
  theme: 'monteCarlo',
  x: xScale,
  y: fn,
  width: 10,
  height: 10,
};

afterEach(cleanup);

test('StackedBarDatum', () => {
  const { container } = render(
    <svg>
      <StackedBarDatum {...props} series={series} />
    </svg>
  );

  // Test just one layer
  const group = container.querySelector('g');
  expect(group).toMatchSnapshot();
  expect(group.classList.contains('bar-layer')).toBe(true);

  // Test just one bar
  const rect = container.querySelector('rect');
  expect(rect).toMatchSnapshot();
  expect(rect.classList.contains('bar')).toBe(true);
});

test('StackedBarDatum:onClick', () => {
  const handleOnClick = jest.fn();
  const { container } = render(
    <svg>
      <StackedBarDatum {...props} series={series} onClick={handleOnClick} />
    </svg>
  );

  const rect = container.querySelector('rect');
  fireEvent.click(rect, {});

  expect(handleOnClick).toHaveBeenCalled();
});

test('StackedBarDatum:onMouseEnter', () => {
  const handleOnMouseEnter = jest.fn();
  const { container } = render(
    <svg>
      <StackedBarDatum
        {...props}
        series={series}
        onMouseEnter={handleOnMouseEnter}
      />
    </svg>
  );

  const rect = container.querySelector('rect');
  fireEvent.mouseEnter(rect, {});

  expect(handleOnMouseEnter).toHaveBeenCalled();
});

test('StackedBarDatum:onMouseLeave', () => {
  const handleOnMouseLeave = jest.fn();
  const { container } = render(
    <svg>
      <StackedBarDatum
        {...props}
        series={series}
        onMouseLeave={handleOnMouseLeave}
      />
    </svg>
  );

  const rect = container.querySelector('rect');
  fireEvent.mouseLeave(rect, {});

  expect(handleOnMouseLeave).toHaveBeenCalled();
});
