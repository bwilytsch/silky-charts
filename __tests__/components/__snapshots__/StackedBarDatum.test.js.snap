// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackedBarDatum 1`] = `
.c0 {
  fill: rgb(8,104,172);
}

.c0:hover {
  fill: rgb(0,84,152);
}

<g
  class="bar-layer"
>
  <rect
    class="bar c0"
    height="10"
    width="0"
    x="0"
    y="0"
  />
  <rect
    class="bar c0"
    height="10"
    width="0"
    x="0"
    y="0"
  />
</g>
`;

exports[`StackedBarDatum 2`] = `
<rect
  class="sc-gzVnrw bar eJUvTq"
  height="10"
  width="0"
  x="0"
  y="0"
/>
`;
